//
// PROJECT:         Aspia
// FILE:            protocol/notifier.proto
// LICENSE:         GNU General Public License 3
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

import "authorization.proto";

package aspia.proto.notifier;

message Session
{
    string uuid                   = 1;
    string remote_address         = 2;
    string username               = 3;
    auth.SessionType session_type = 4;
}

message SessionClose
{
    string uuid = 1;
}

message KillSession
{
    string uuid = 1;
}

message NotifierToService
{
    KillSession kill_session = 1;
}

message ServiceToNotifier
{
    Session session            = 1;
    SessionClose session_close = 2;
}
